### ISSUE TITLE
Support IPv6

### ISSUE THREAD

**Issue by svetlyak40wt** [fukamachi/woo#53]
Is it hard to support binding to IPv6 in Woo?

I found woo is the only handler for Clack which accepts `:address` argument, but when I pass "::0" it is failing. And looking [at the code](https://github.com/fukamachi/woo/blob/master/src/ev/tcp.lisp#L223-L227) seems it only support IPv4 for now.

I have a bunch os IPv6 only backends and want to run `ningle` based microservice on them. Having support for IPv6 in Woo whould be wonderful!

P.S. — found a great tutorial [Porting applications to IPv6 HowTo ](http://long.ccaba.upc.es/long/045Guidelines/eva/ipv6.html). May be will be able to fix woo myself. Anyway, any guidance from you will be highly appreciated! Thank you for great libraries!

**Comment by svetlyak40wt** [comment]
Found a workaround – start an woo server on a unix socket and proxy to it from local Nginx. But it doesn't work. Unix socket is created, but I make request to `nginx`, `woo` server writes to stdout:
```
<ERROR> [08:18:32] woo.ev.tcp - Can't accept connection (Code: 14)
```
and nginx returns 502 Bad Gateway, outputting to logs:
```
2016/08/06 08:22:22 [error] 60#0: *1 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.17.0.1, server: , request: "GET /info HTTP/1.1", upstream: "http://unix:/app.sock:/info", host: "localhost:8000"
172.17.0.1 - - [06/Aug/2016:08:22:22 +0000] "GET /info HTTP/1.1" 502 181 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/601.6.17 (KHTML, like Gecko) Version/9.1.1 Safari/601.6.17"
```

Seems, error 14, mentioned by `woo`, is:
```
#define EFAULT      14  /* Bad address */
```

Do you have any ideas what is wrong and how to fix it?

**Comment by fukamachi** [comment]
The EFAULT error is because the `sockaddr_in` is short for IPv6 addresses.
Replacing it by `sockaddr_in6` or `sockaddr_storage` might work.

**Comment by svetlyak40wt** [comment]
By the way, workaround with unix-socket worked after patch from @knobo was applied. I started [supervisord](http://supervisord.org) and lisp + nginx under it. Nginx listens on [::]80 and proxies to unix-socket.

But I feel something wrong to have such configuration inside a docker container.

**Comment by fukamachi** [comment]
Implemented at #55.
I'll merge it after some benchmarks and performance improvements.

@svetlyak40wt Could you give a try the ipv6 branch?

**Comment by svetlyak40wt** [comment]
Sure.
